lib/streamlit/commands/navigation.py

```diff
@@ -0,0 +1,189 @@
+    msg.page_not_found.page_name = ""
+    ctx.enqueue(msg)
+
+
+@gather_metrics("navigation")
+def navigation(
+    pages: list[StreamlitPage] | dict[SectionHeader, list[StreamlitPage]],
+    *,
+    position: Literal["sidebar"] | Literal["hidden"] = "sidebar",
+) -> StreamlitPage | None:
+    """
+    Configure the available pages in a multipage app.
+
+    Call `st.navigation` in your main script with one or more pages defined by
+    `st.Page`. `st.navigation` returns the current page which can be executed
+    using `Page.run()`.
+
+    When using `st.navigation`, the main script is executed on every app rerun,
+    with the current page executed in-line using `Page.run()`. The set of
+    available pages can be updated with each rerun for dynamic navigation. In
+    this mode, the `pages/` folder is ignored.
+
+    By default, `st.navigation` draws the available pages in the side
+    navigation. This behavior can be changed using the `position=` keyword
+    argument.
+
+    Parameters
+    ----------
+    pages: list[st.Page] or dict[str, list[st.Page]]
+        A list of `st.Page` objects or a dictionary where the keys are section
+        headers and the values are lists of `st.Page` objects.
+
+    position: str
+        The position of the navigation menu. Can be "sidebar" or "hidden".
+
+    Example
+    -------
+    >>> import streamlit as st
+    >>> from pages import page1, page2
+    >>>
+    >>>	pg = st.navigation([st.Page(page1), st.Page(page2)])
+    >>>
+    >>>	st.title("My Awesome App")
+    >>> pg.run()
+    """
+    ctx = get_script_run_ctx()
+    if not ctx:
+        return None
+
+    nav_sections = {"": pages} if isinstance(pages, list) else pages
+    page_list = pages_from_nav_sections(nav_sections)
+
+    if not page_list:
+        raise StreamlitAPIException(
+            "`st.navigation` must be called with at least one `st.Page`."
+        )
+
+    default_page = None
+    pagehash_to_pageinfo: dict[PageHash, PageInfo] = {}
+
+    # This nested loop keeps track of three things:
+    # 1. the default page
+    # 2. the pagehash to pageinfo mapping
+    for section_header in nav_sections:
+        for page in nav_sections[section_header]:
+            if page.default:
+                if default_page is not None:
+                    raise StreamlitAPIException(
+                        "Multiple Pages specified with `default=True`. "
+                        "At most one Page can be set to default."
+                    )
+                default_page = page
+
+            if isinstance(page._page, Path):
+                script_path = str(page._page)
+            else:
+                script_path = ""
+
+            script_hash = page._script_hash
+            if script_hash in pagehash_to_pageinfo:
+                raise StreamlitAPIException(
+                    f"Multiple Pages specified with title {page.title}. "
+                    "Page titles should be unique. If not specified, "
```

Comment:
nit: "Page titles should be unique" -> "Page titles must be unique"

Reply:
I think this is a weird error. The error case is: "when there's duplicate script hashes these two pages share all similar information" The way to differentiate would be the title. So not every page title HAS to be unique, but if there's no other way to define the script hash, the titles need to be changed.

In fairness, this should be a relatively rare exception.